app/library-proxy.hoon

```diff
@@ -0,0 +1,467 @@
+/-  *resource, library
+/+  store=graph-store, graph, default-agent,
+    dbug, verb, agentio, libr=library
+::  TODO i still don't know what wires to put for my `%watch`s
+::  TODO should i refactor out %give paths to arm (i.e. generate /updates/(scot %p her)/(scot %p our.bowl)/[name.rid] from arm)
+|%
++$  versioned-state
+    $%  state-0
+    ==
+::
++$  state-0  [%0 base-state-0]
++$  base-state-0
+  $:
+    =readers:library
+    =policies:library
+  ==
+::
++$  card  card:agent:gall
+--
+%-  agent:dbug
+=|  state-0
+=*  state  -
+=<
+^-  agent:gall
+|_  =bowl:gall
++*  this        .
+    def       ~(. (default-agent this %|) bowl)
+    hc        ~(. +> bowl)
+    io        ~(. agentio bowl)
+    gra       ~(. graph bowl)
+::
+++  on-init
+  ^-  (quip card _this)
+  ~&  >  '%library-proxy initialized successfully'
+  [(~(watch-our pass:io /local-store) %graph-store /updates)^~ this]
+++  on-save
+  ^-  vase
+  !>(state)
+++  on-load
+  |=  old-state=vase
+  ^-  (quip card _this)
+  ~&  >  '%library-proxy recompiled successfully'
+  `this(state !<(versioned-state old-state))
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  =^  cards  state
+  ?+    mark  (on-poke:def mark vase)
+      %library-command
+    ::?>  (team:title our.bowl src.bowl)  :: allow ourselves and moons to use this poke
```

Comment:
I think we're moving away from moons acting as equivalent to the planet in terms of permissions for 3rd party software distribution, probably get rid of this

Reply:
https://github.com/ynx0/library/pull/3/commits/1fdca233683c347287800451016bfae4128ba125